### PR TITLE
Block the opening of potentially malicious URLs via custom intents su…

### PR DESCRIPTION
**Pull request by Brandon-T**
## Summary
- Block the opening of potentially malicious URLs via custom intents such as Shortcuts (Bookmarks, History, Open Website, etc).

## Note to Reviewers
- We do not block the saving of `file://` URLs because downloading PDFs are stored to `/tmp` and saved from there. We could technically validate that the `Save to Files` option saves from` /tmp`, but not sure if it's necessary since we block the opening of `file://` anyway.

Resolves https://github.com/brave/reviews/issues/1686

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: ios/brave-ios/Sources/BrowserIntentsModels/CustomIntentHandler.swift [hunk 1]
```diff
@@ -7,26 +7,44 @@ import CoreSpotlight
 import Intents
 import MobileCoreServices
 
+private func isWebPage(url: URL?) -> Bool {
```
Comment (iccub) [inline ios/brave-ios/Sources/BrowserIntentsModels/CustomIntentHandler.swift:10]: There is a `isWebPage` method in Shared's URLExtension class. Should we merge or update them or is it different?

Comment (Brandon-T) [reply ios/brave-ios/Sources/BrowserIntentsModels/CustomIntentHandler.swift:10]: This is inside of custom intents. I'd have to import those extensions, which I'm not sure if that's possible for donated intents which run out of process in the Shortcuts app. I'll have to check.

Comment (iccub) [reply ios/brave-ios/Sources/BrowserIntentsModels/CustomIntentHandler.swift:10]: okay please check, this is a non-blocking comment

**Review by iccub — CHANGES_REQUESTED** [review]
CI does not like this change
```
[2024-07-13T15:43:26.250Z] ❌ /Users/jenkins/jenkins/workspace/brave-core-build-pr-ios_PR-24640/src/brave/ios/brave-ios/Sources/BrowserIntentsModels/CustomIntentHandler.swift:9:8: no such module 'Shared'
```

and you can check if `testisWebPage` test needs an update

**Comment by Brandon-T** [comment]
> CI does not like this change
> 
> ```
> [2024-07-13T15:43:26.250Z] ❌ /Users/jenkins/jenkins/workspace/brave-core-build-pr-ios_PR-24640/src/brave/ios/brave-ios/Sources/BrowserIntentsModels/CustomIntentHandler.swift:9:8: no such module 'Shared'
> ```
> 
> and you can check if `testisWebPage` test needs an update

Weird. It's compiling and running locally :S

**Review by iccub — APPROVED** [review]

**Review by stoletheminerals — APPROVED** [review]
I didn't test dynamically but the code lgtm

**Pull request merged**